server/routes/api/place.py

```diff
@@ -800,30 +801,50 @@ def api_ranking_chart(dcid):
         parent_place_dcid = EARTH_DCID
         place_type = "Country"
     else:
-        parent_place_list = get_parent_place(dcid).get(dcid, [])
-        # Get the last parent place returned.
-        if parent_place_list:
-            parent_place_dcid = parent_place_list[-1]["dcid"]
-            place_type = get_place_type(dcid)
+        place_type = get_place_type(dcid)
+        parent_place_list = parent_places(dcid).get(dcid, [])
+        for parent in parent_place_list:
+            parent_type_list = parent.get("types", [])
+            parent_place_dcid = parent.get("dcid", "")
+            if parent_type_list:
+                parent_type = parent_type_list[0]
+                # All wanted place types plus continent except CensusZipCodeTabulationArea.
+                if parent_type == "Continent" or (
+                        parent_type in ALL_WANTED_PLACE_TYPES and
+                        parent_type != "CensusZipCodeTabulationArea"):
+                    break
+        # If break is not encountered, return empty result.
         else:
             return Response(json.dumps(result),
                             200,
                             mimetype='application/json')
+
     configs = get_ranking_chart_configs()
     # Get the first stat var of each config.
     stat_vars, _ = shared_api.get_stat_vars(configs)
+    # Make sure the ranking chart configs include stat var "Count_Person".
+    if "Count_Person" not in stat_vars:
+        stat_vars.add("Count_Person")
     sv_data = dc.get_stat_set_within_place(parent_place_dcid, place_type,
```

Comment:
We are trying to migrate to V1 APIs (from Mixer) now & to help with this process, let's try to use the new mixer APIs wherever possible in new and touched code. I have a [pr](https://github.com/datacommonsorg/website/pull/1638/files) up right now that adds the new function dc.points_within that will eventually replace dc.get_stat_set_within_place. Once that pr is in, let's update this flask endpoint to use the new function. Can do this in a followup PR

Reply:
Sure!